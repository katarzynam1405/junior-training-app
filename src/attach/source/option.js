export default function attachSourceOption(sourceList, dataSourceAttr) {
  document.querySelector(`[${dataSourceAttr}]`).innerHTML = sourceList.join('');
}
