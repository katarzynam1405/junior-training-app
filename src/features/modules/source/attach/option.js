/**
 * changes HTML structure by adding array
 * @param {array} sourceList
 * @param {Object} dataSourceAttr
 */
export default function attachSourceOption(sourceList, dataSourceAttr) {
    document.querySelector(dataSourceAttr).innerHTML = sourceList.join('');
}
