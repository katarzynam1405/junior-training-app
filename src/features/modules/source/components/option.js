// eslint-disable-next-line
import React from 'react';
/**
 * changes list of objects to list of strings that can be used in html tree
 * @param {array} sources
 * @param {string} template
 */

export default function SourceOption(sources) {
    return (<select>{sources.map(option => <option>{option}</option>)}</select>);
}
