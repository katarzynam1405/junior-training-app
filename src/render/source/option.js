export default function renderSourceOption(sources, template) {
  const result = [];

  const replacement = { key: 'index', value: 'source' };
  const getValue = (tag, source) => {
    let val = '';
    for (const key in replacement) {
      if (key === tag) {
        val = source[replacement[key]];
        break;
      }
    }
    return val;
  };

  sources.forEach((source, index) => result.push(template.replace(/{(\w+)}/g,
    (match, tag) => getValue(tag, { source, index }))));

  return result;
}
