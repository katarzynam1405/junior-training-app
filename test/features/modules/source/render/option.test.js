import option from "../../../../../src/features/modules/source/render/option"

describe('render/source/option', () => {
    it('should match to Snapshot', () => {
        
        const results = option([
            {index: "qwer", source: "1234"},
            {index: "rock", source: "0987"}
        ], "lorem_{key}_{value}")
        expect(results).toMatchSnapshot();
    })
});
