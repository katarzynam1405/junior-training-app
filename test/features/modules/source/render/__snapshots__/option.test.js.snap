// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`f/m/source/render/option should match to Snapshot 1`] = `
Array [
  "lorem_0_[object Object]",
  "lorem_1_[object Object]",
]
`;
